akka-http/src/main/scala/akka/http/scaladsl/server/HttpApp.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.http.scaladsl.server
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.{ Http, server }
+import akka.stream.ActorMaterializer
+import akka.http.scaladsl.settings.ServerSettings
+
+import scala.io.StdIn
+
+/**
+ * Bootstrap trait. It helps booting up an akka-http by only defining the desired routes.
+ * It offers additional hooks to modify the default behavior
+ */
+trait HttpApp {
+
+  def main(args: Array[String]) = {
+    implicit val system = actorSystem()
+    implicit val materializer = ActorMaterializer()
+    implicit val executionContext = system.dispatcher
+    postStartActorSystem(system)
+
+    val bindingFuture = Http().bindAndHandle(
+      handler = route,
+      interface = config.host,
+      port = config.port,
+      settings = config.serverSettings.getOrElse(ServerSettings(system)))
+
+    println(s"Server online at http://${config.host}:${config.port}/\nPress RETURN to stop...")
+    StdIn.readLine()
```

Comment:
Can you make this overridable? I imagine some implementations wanting to wait on a different signal.